actionpack/lib/action_dispatch/http/request.rb

```diff
@@ -393,8 +388,6 @@ def POST
         pr = parse_formatted_parameters(params_parsers) do |params|
           super || {}
         end
-        pr = Request::Utils.set_binary_encoding(self, pr, path_parameters[:controller], path_parameters[:action])
-        Request::Utils.check_param_encoding(pr)
         self.request_parameters = Request::Utils.normalize_encode_params(pr)
```

Comment:
We might be able to move setting the request parameters here (same thing with `self.query_parameters = Request::Utils.normalize_encode_params(rack_query_params)` above in `GET`) now that we're setting them in the router?